### DISCUSSION TITLE
Get directed to different page

### DISCUSSION THREAD

**Discussion by SelcoD** [neuefische/questions#636] [Web] [answered]
Can anybody help me with this ?
I want to click on a card at the listpage and want to get directed to a detailed site of the trip

For demonstation how the detailpage/1 should look like i did not delete the DetailPage for now, later i will delete it.

Anybody an idea how i can cklick one of the cards in the listpage an get directed to the /detailpage/1
with the informations from the trip ?

I am stuck at this point.

[PR] (https://github.com/SelcoD/CampingFriends/pull/6)

[US 2] (https://github.com/users/SelcoD/projects/1/views/1?pane=issue&itemId=32603563)

**Comment by doemser** [comment]
Sounds like you want to implement dynamic routes! 

Have you looked at the [next.js dymanic routes - Handout](https://github.com/neuefische/bo-web-23-3/blob/main/sessions/nextjs-dynamic-routes/nextjs-dynamic-routes.md) in your course repository yet? How to handle dynamic routes is covered there.

Your Markdown links are broken, here is how to write them. 
```
[Link Title](https://url.you.want)
```

**Comment by SelcoD** [comment]
Yes i read this Handout but i cant find the problem.

**Comment by SelcoD** [comment]
<img width="847" alt="Bildschirmfoto 2023-07-11 um 16 19 37" src="https://github.com/neuefische/questions/assets/132760864/1f4c2e3a-7426-4dd4-8a92-873928db6f28">
<img width="603" alt="Bildschirmfoto 2023-07-11 um 16 19 55" src="https://github.com/neuefische/questions/assets/132760864/bd999895-1134-4d62-8218-30562c014053">

Also er empfängt den trip irgendwie nicht.
Ich hab nun in Zeile 49 den trip loggen lassen und es erscheint undefined.

**Comment by actyvystom** [comment, marked as answer]
The problem is differing data types. In your camping trip data, you use a number as id and in your dynamic detail route your id is a string by default.
```js
...
{
    id: 1,
    day: "Monday",
    date: "6.03.2023",
    location: "Leucate",
    details: ["-windy", "-large spaces", "-sunny", "-cloudy"],
    friends: ["Chris", "Maik"],
    images: ["/../public/leucate-pano.jpg", "/../public/leucate-plage.jpeg"],
  },
...
``` 
Either convert your `id` retrieved from `router.query` to a number or (better) use strings for your ids (e.g. by using an id generator package) or simply by manually changing your trip data ids to string:
```js
...
{
    id: "1",
    day: "Monday",
    date: "6.03.2023",
    location: "Leucate",
    details: ["-windy", "-large spaces", "-sunny", "-cloudy"],
    friends: ["Chris", "Maik"],
    images: ["/../public/leucate-pano.jpg", "/../public/leucate-plage.jpeg"],
  },
...
```

**Comment by SelcoD** [comment]
Hi, yes i fixed it.
<img width="784" alt="Bildschirmfoto 2023-07-11 um 17 00 30" src="https://github.com/neuefische/questions/assets/132760864/4131fc2a-a4ce-472c-bb5f-06862af31e5b">
 now it works.
 
 Thanks for helping
